### ISSUE TITLE
Major bug when switching esri dynamic legend item visibility

### ISSUE THREAD

**Issue by jolevesq** [fgpv-vpgf/fgpv-vpgf#4047]
When we add these keys in RAMP 3.3.6:

f4c51eaa-a6ca-48b9-a1fc-b0651da20509%2c65d3db23-b83c-4f49-ab93-65c59ee0e6aa%2cd256b422-2834-40a2-9f0c-bd5fc32781b2%2c21b821cf-0f1c-40ee-8925-eab12d357668

and we turn the visibility of sub layers item, the layers are not able to load and the UI tries to refresh itself in an endless loop. In version 3.3.5, the layer fails to load but the ui is ok.

![RAMP-3 3 6](https://user-images.githubusercontent.com/3472990/193055364-725a34e6-34a9-46fa-9c9c-f350fc7ec950.PNG)

Can you please test on your side to let us know if you can reproduce. Thanks

**Comment by james-rae** [comment]
We don't know what those keys are. Can you paste the layer config JSON objects that those keys map to?

**Comment by jolevesq** [comment]
It is RCS keys in OpensMaps. It give back this configuration:

```
[
{
"layers": [
{
"isTimeAware": false,
"layerType": "esriDynamic",
"service_url": "https://maps-cartes.services.geo.ca/server_serveur/rest/services/IAAC/assessment_inventory_en/MapServer",
"name": "Impact Assessment Agency of Canada - Assessments Inventory",
"id": "rcs.f4c51eaa-a6ca-48b9-a1fc-b0651da20509.en",
"url": "https://maps-cartes.services.geo.ca/server_serveur/rest/services/IAAC/assessment_inventory_en/MapServer",
"layerEntries": [
{
"index": 0
},
{
"index": 1
},
{
"index": 2
}
]
}
]
},
{
"layers": [
{
"isTimeAware": false,
"layerType": "esriDynamic",
"service_url": "https://geoappext.nrcan.gc.ca/arcgis/rest/services/Energy/CERP_EN/MapServer",
"name": "Clean power generating stations by type in megawatts (MW)",
"id": "rcs.65d3db23-b83c-4f49-ab93-65c59ee0e6aa.en",
"url": "https://geoappext.nrcan.gc.ca/arcgis/rest/services/Energy/CERP_EN/MapServer",
"layerEntries": [
{
"index": 0
}
]
}
]
},
{
"layers": [
{
"isTimeAware": false,
"layerType": "esriDynamic",
"service_url": "https://maps-cartes.services.geo.ca/server_serveur/rest/services/HC/stations_en/MapServer",
"name": "Monitoring Stations",
"id": "rcs.d256b422-2834-40a2-9f0c-bd5fc32781b2.en",
"url": "https://maps-cartes.services.geo.ca/server_serveur/rest/services/HC/stations_en/MapServer",
"layerEntries": [
{
"index": 0
}
]
}
]
},
{
"layers": [
{
"isTimeAware": false,
"layerType": "esriDynamic",
"service_url": "https://maps-cartes.services.geo.ca/server_serveur/rest/services/HC/airborne_radioactivity_en/MapServer",
"name": "Airborne Radioactivity",
"id": "rcs.21b821cf-0f1c-40ee-8925-eab12d357668.en",
"url": "https://maps-cartes.services.geo.ca/server_serveur/rest/services/HC/airborne_radioactivity_en/MapServer",
"layerEntries": [
{
"index": 0
}
]
}
]
}
]
```

**Comment by james-rae** [comment]
We're attempting debugs, but are noticing a lot of strange returns from this map server. The server appends two `access-control-allow-origin` headers, one with `*`, another with our domain, which is a security violation of sorts to the browser and it throws a CORS error. It doesn't appear to be hitting you (as you screenshot shows it load much father past that point). Curious if you have any info on whats up with that server and the headers?

We'll keep prodding regardless...just tricky when no layer loads.

**Comment by jolevesq** [comment]
You are right, this server is tricky. It is an old server who is suppose to be decommission soon. I will contact someone from server team to investigate. Thanks

**Comment by james-rae** [comment]
Was able to create a working sample that has similar scenarios and appears to be working ok. [Demo link](https://fgpv-vpgf.github.io/fgpv-vpgf/coulson/samples/index-fgp-en.html?keys=f4c51eaa-a6ca-48b9-a1fc-b0651da20509)

This loads stuff via the RCS-style url `keys` array. It has two payloads, one of your layers provided above, and one from our servers (in an attempt to see if the CORS stuff was the root of the problem).

Things appear to be working as they should on our sample. So we may need to do some environmental difference testing? 
Also i see in your screenshot you are using a local host build. If that is not `https`, our team has noticed that using services from your `.geo.ca` map server we get lots of problems now under `http` so maybe you want to try something in an `https` environment?

**Comment by jolevesq** [comment]
@james-rae Thanks!
